nvflare/fuel/f3/cellnet/new_cell.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import threading
+import uuid
+from typing import Dict, List, Union
+
+from nvflare.fuel.f3.cellnet.cell import Cell
+from nvflare.fuel.f3.cellnet.defs import MessageHeaderKey, MessageType
+from nvflare.fuel.f3.message import Message
+from nvflare.fuel.f3.stream_cell import StreamCell
+from nvflare.fuel.f3.streaming.stream_const import StreamHeaderKey
+from nvflare.fuel.f3.streaming.stream_types import StreamFuture
+from nvflare.private.defs import CellChannel
+
+
+class Adapter:
+    def __init__(self, cb, my_info, nice_cell):
+        self.cb = cb
+        self.my_info = my_info
+        self.nice_cell = nice_cell
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    def call(self, future):  # this will be called by StreamCell upon receiving the first byte of blob
+        headers = future.headers
+        stream_req_id = headers.get(StreamHeaderKey.STREAM_REQ_ID, "")
+        origin = headers.get(MessageHeaderKey.ORIGIN, None)
+        result = future.result()
+        request = Message(headers, result)
+        channel = request.get_header(StreamHeaderKey.CHANNEL)
+        request.set_header(MessageHeaderKey.CHANNEL, channel)
+        topic = request.get_header(StreamHeaderKey.TOPIC)
+        request.set_header(MessageHeaderKey.TOPIC, topic)
+        req_id = request.get_header(MessageHeaderKey.REQ_ID)
+        response = self.cb(request)
+        response.add_headers(
+            {
+                MessageHeaderKey.REQ_ID: req_id,
```

Comment:
Will `req_id` be None? If so, do we want to pass it here?

Reply:
The default is an empty string (like in cell.py case).  If the receiving side can't find the matching req_id, it will ignore that reply.  I think reply arriving too late is one case.